crates/sozo/ops/src/migration/auto_auth.rs

```diff
@@ -0,0 +1,63 @@
+use anyhow::Result;
+use dojo_world::contracts::{cairo_utils, WorldContract};
+use dojo_world::manifest::BaseManifest;
+use dojo_world::migration::TxnConfig;
+use scarb::core::Workspace;
+use scarb_ui::Ui;
+use starknet::accounts::ConnectedAccount;
+
+use super::ui::MigrationUi;
+use super::MigrationOutput;
+use crate::auth::{grant_writer, ModelContract};
+
+pub async fn auto_authorize<'a, A>(
+    ws: &Workspace<'_>,
+    world: &'a WorldContract<A>,
+    txn_config: &TxnConfig,
+    local_manifest: &BaseManifest,
+    migration_output: &MigrationOutput,
+) -> Result<()>
+where
+    A: ConnectedAccount + Sync + Send + 'static,
+{
+    let ui = ws.config().ui();
+
+    ui.print(" ");
+    ui.print_step(6, "🖋️", "Authorizing Models to Systems (based on overlay)...");
+    ui.print(" ");
+    let models_contracts = compute_models_contracts(&ui, local_manifest, migration_output)?;
+    grant_writer(&ui, world, models_contracts, *txn_config).await
+}
+
+pub fn compute_models_contracts(
+    ui: &Ui,
+    local_manifest: &BaseManifest,
+    migration_output: &MigrationOutput,
+) -> Result<Vec<crate::auth::ModelContract>> {
+    let mut res = vec![];
+    let local_contracts = &local_manifest.contracts;
+
+    // from all the contracts that where migrated successfully
```

Comment:
```suggestion
    // From all the contracts that were migrated successfully.
```
Trying to keep comment as punctuated sentences when possible.